src/weakptr.c

```diff
@@ -636,7 +636,7 @@ static Obj CopyObjWPObj(Obj obj, Int mut)
         ADDR_OBJ(copy)[0] = CONST_ADDR_OBJ(obj)[0];
     }
     else {
-        copy = NewBag( T_PLIST+IMMUTABLE, SIZE_OBJ(obj) );
+        copy = NEW_PLIST( T_PLIST+IMMUTABLE, LengthWPObj(obj) );
         SET_LEN_PLIST(copy,LengthWPObj(obj));
```

Comment:
Note that the value of `LengthWPObj(obj)` could change between the two previous lines. And the loop a few lines later again uses `SIZE_OBJ(obj)`...